staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go

```diff
@@ -18,15 +18,30 @@ package legacyregistry
 
 import (
 	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
+
 	apimachineryversion "k8s.io/apimachinery/pkg/version"
 	"k8s.io/component-base/metrics"
-	"sync"
 )
 
-var globalRegistryFactory = metricsRegistryFactory{
-	registerQueue:     make([]metrics.KubeCollector, 0),
-	mustRegisterQueue: make([]metrics.KubeCollector, 0),
-}
+var (
+	globalRegistryFactory = metricsRegistryFactory{
+		registerQueue:     make([]metrics.KubeCollector, 0),
+		mustRegisterQueue: make([]metrics.KubeCollector, 0),
+		globalRegistry:    noopRegistry{},
```

Comment:
Actually I don't really get why we need such `noopRegistry` here.
And I prefer not to preset `globalRegistryFactory` with `noopRegistry`, since functions like `SetRegistryFactoryVersion` will make changes to `globalRegistryFactory` as well.

Reply:
I personally prefer the explicitness of overriding a default noop registry. Otherwise, the behavior is dependent on implications of having a global registry version registered.

Reply:
It's also more consistent with other usages in Kubernetes: https://github.com/kubernetes/kubernetes/blob/61fa0fd44a3513b94d7c29ff859340804c107ea0/staging/src/k8s.io/client-go/tools/cache/reflector_metrics.go#L77.